### PR TITLE
Fix the problem that ttlManager may stop working if no need to lock after retry aggressive locking

### PR DESCRIPTION
**Pull request by MyonKeminta**
Ref: https://github.com/pingcap/tidb/issues/58279

### CODE REVIEW

File: txnkv/transaction/txn.go [hunk 1]
```diff
@@ -1075,6 +1079,12 @@ func (txn *KVTxn) DoneAggressiveLocking(ctx context.Context) {
 		txn.aggressiveLockingContext = nil
 	}()
 
+	// If finally no key locked and ttlManager is just started during the current fair locking procedure, reset the
+	// ttlManager as no key will be the primary.
+	if txn.aggressiveLockingContext.lastAssignedPrimaryKey && !txn.aggressiveLockingContext.assignedPrimaryKey {
+		txn.committer.ttlManager.reset()
```
Comment (cfzjywxk) [inline txnkv/transaction/txn.go:1085]: Should we return here if finally no key is locked?

Comment (MyonKeminta) [reply txnkv/transaction/txn.go:1085]: Better not I think. There are cleanup steps and maxLockedWithConflictTS calculation later.

File: txnkv/transaction/txn.go [hunk 2]
```diff
@@ -1559,9 +1571,37 @@ func (txn *KVTxn) lockKeys(ctx context.Context, lockCtx *tikv.LockCtx, fn func()
...
+		return
+	}
+	// * When there's no new lock to acquire, assume the primary key is not changed in this case. Keep the ttlManager
+	// running.
+	// * When there is key to write:
+	//   * If the primary key is not changed, also keep the ttlManager running. Then, when sending the
+	//     acquirePessimisticLock requests, it will call ttlManager.run() again, but it's reentrant and will do nothing
+	//     as the ttlManager is already running.
+	//   * If the primary key is changed, the ttlManager will need to run on the new primary key instead. Reset it.
+	if !noNewLockToAcquire && !bytes.Equal(txn.aggressiveLockingContext.primaryKey, txn.aggressiveLockingContext.lastPrimaryKey) {
```
Comment (ekexium) [inline txnkv/transaction/txn.go:1602]: Maybe replacing `!noNewLockToAcquire` with `hasNewLockToAcquire` is more idiomatic and makes the code more readable

**Review by ekexium — APPROVED** [review]

File: txnkv/transaction/txn.go (hunk 2 above)
Comment (MyonKeminta) [reply txnkv/transaction/txn.go:1602]: emmmm, ok

**Review by cfzjywxk — APPROVED** [review]
LGTM

**Pull request merged**